### ISSUE TITLE
bulk delete and bulk move in bookmarks a future feature?

### ISSUE THREAD

**Issue by roo-the-kangaroo** [organicmaps/organicmaps#6675]
Is bulk delete and bulk move in bookmarks a future feature? If not, it should be.
Deleting several bookmarks within a list at once, or move several bookmarks from one list to another, would be neat.

_Originally posted by @roo-the-kangaroo in https://github.com/organicmaps/organicmaps/issues/644#issuecomment-1820350678_

**Comment by sklueber** [comment]
What's the expected UX here? Based on Android conventions I would expect a long press on a bookmark to start the selection and then using the overflow menu for bulk delete and move (+ edit color & share?).

However, right now a long press on a bookmark brings up options for the bookmark. Would it be acceptable to move these into the overflow menu as well? Edit would only show up when only one bookmark is selected. That's the way it works in the Google file manager for example. 
The edit icon could also be in the action bar (replacing the search icon during selection) to make it more accessible.

**Comment by roo-the-kangaroo** [comment]
Yes, that sounds good.
A long press on a bookmark to start the selection and then use the overflow menu to delete, move, edit the colour or split.
Yes, but one of the options could be the selection. And when a bookmark is selected, others can be selected with one click.
But for me, it is also ok to move them to the overflow menu if this is easier.

**Comment by biodranik** [comment]
It would be great to see some proposed approaches on recorded screen videos.

@sandy99405 thanks for the video, your approach has some inconsistencies that can be improved/resolved.

Note that bookmarks may also display a distance to them. Is there an issue about it already?

**Comment by sandy99405** [comment]
@biodranik  thank you for pointing it out 😊. You are right, there are still some inconsistencies in the output. I've faced many and solved them as well. This is still under development. I will get back to you with 0 inconsistencies.👍

**Comment by sandy99405** [comment]
@biodranik the distance not showing bug occurred before I introduced the checkboxes. Do you want me to raise a new issue about it?

**Comment by sklueber** [comment]
Is it possible that you simply have turned off GPS on your phone? Then there are no distances (because what distance would it show?)

**Comment by sandy99405** [comment]
@biodranik  everything is working fine with the checkboxes. Shall I create a PR now?

**Comment by biodranik** [comment]
Before proceeding, can you please record some videos on how bulk actions are implemented in Android system apps and the most popular apps? Is it done similarly on recent Androids?

**Comment by sandy99405** [comment]
@biodranik  I get that you guys asked for a long press of the bookmark and then the start of bookmark selection. But the long press is already used for bottom menu sheet so I maintained the checkboxes directly.

**Comment by sandy99405** [comment]
The long press selection implementation is ready as well. Would you like me to give a demo of it?

**Comment by sandy99405** [comment]
Hi @biodranik here is the long press checkbox implementation.

https://github.com/organicmaps/organicmaps/assets/55664068/07e984fa-e315-4ba5-8de8-d98b886e693a

Is this fine ?

**Comment by biodranik** [comment]
I'm trying to understand first, what is the best Android way to implement it. How other apps behave in similar situations. We need it to be familiar to users, and it should take less clicks if possible.

**Comment by sandy99405** [comment]
@biodranik what I have observed in most of the apps is:
The user long clicks on an item and starts selecting each item to perform an action(moving, copying etc).

In the above implementation : 
The user long clicks and the checkboxes appear for him/her to select the items and then the overflow drop down button will enable the bulk actions feature.


Which one do you think is best for your app ? I will do as you say.

**Comment by sandy99405** [comment]
Hi @biodranik shall I implement the long click and selecting the bookmarks with click ?

**Comment by biodranik** [comment]
@euf requesting your help )

**Comment by sandy99405** [comment]
@euf for bulk moving and deletion of bookmarks, I am suggesting for the checkbox appearance followed by long press. But most of the android apps use selection of items by a click followed by long press on the item. So which one do you want me to implement? I need your help please. You can even watch the video that I have shared in this chain as well.

**Comment by euf** [comment]
@sandy99405 Thank you for explaining me this two possible approaches! Please give me a couple of days to consider these options and draw a conclusion based on the current platform guidelines / best practices. Manipulating lists on mobile is always challenging from the UX perspective 🙏

**Comment by euf** [comment]
Hi, just a heads up: I think I'm really close to figuring out the right UX for bulk editing on Android. This might possibly require a little more work on restructuring the interface elements, but the idea of long press stays the same. I'm just trying to make sure that the UI would be future proof for the further development so that users wouldn't have to learn things twice. Please give me just a few more days before I'm ready to share some screenshots 🙏

**Comment by hoizan** [comment]
I am looking forward to this bulk delete/move feature. This PRl looks so promising, but wondered  why it seems stalled.

**Comment by sandy99405** [comment]
> I am looking forward to this bulk delete/move feature. This PRl looks so promising, but wondered why it seems stalled.

Hello @hoizan . I have been looking forward to work on this too. But @euf has been doing some analysis and collecting some data from different android apps for this feature. So as soon as we get some information and acceptance, I will start to create a PR for this feature. Thank you.

**Comment by euf** [comment]
Hello, sorry for keeping you wait so long. 

Here's what I came to: 

- Long tap on an item in the list should initiate "Edit mode"
- The item which received the long tap should be already selected with a checkbox
- In this edit mode user can select any amount of items, from zero to all of them 
- The bulk operations are available on selected items (move to another list, change color, delete)
- Move operation should result in navigating to a new location
- The edit mode should be visually distinct from a regular mode (e.g. dark colored header with "X selected" text) 
- Leaving edit mode should just keep the list at the same position it was before

Please also check a suggested design for new list and item management: 
https://www.figma.com/design/xrgJAfVqw0HNkPiLmE3TXF/OM-Android-Bookmarks-%26-Tracks

It was sketched up supposing that bookmarks menu would be eventually moved to the bottom sheet. This can surely be achieved by increments, going step by step, and bulk editing would be a great start!

**Comment by roo-the-kangaroo** [comment]
Yes, that's fine with me. The bookmark features that would be great would
be: delete, move, copy & edit the colour. It would be cool to have not only
bulk features for the bookmarks, but also for the lists.

**Comment by roo-the-kangaroo** [comment]
Any update on this feature? Waiting for ages for this feature. I download the GPX file from Wikivoyage when I travel, but waste time in deleting hotels and other bookmarks I don't need.
